src/addresses.ts

```diff
@@ -0,0 +1,233 @@
+import { ChainId, SUPPORTED_CHAINS, SupportedChainsType } from './chains'
+
+type AddressMap = { [chainId: number]: string }
+
+type ChainAddresses = {
+  v3CoreFactoryAddress: string
+  multicallAddress: string
+  quoterAddress: string
+  v3MigratorAddress?: string
+  nonfungiblePositionManagerAddress?: string
+  tickLensAddress?: string
+  swapRouter02Address?: string
+  v1MixedRouteQuoterAddress?: string
+}
+
+const DEFAULT_NETWORKS = [ChainId.MAINNET, ChainId.GOERLI]
+
+function constructSameAddressMap(address: string, additionalNetworks: ChainId[] = []): AddressMap {
+  return DEFAULT_NETWORKS.concat(additionalNetworks).reduce<AddressMap>((memo, chainId) => {
+    memo[chainId] = address
+    return memo
+  }, {})
+}
```

Comment:
I find the `DEFAULT_NETWORKS` piece a bit confusing personally--I know you've copied it from interface and divergent implementations would be a little annoying, but requiring explicit enumeration of the full set at runtime feels more predictable to me.

Reply:
Do you think the added clarity is worth the tradeoff of the redundancy of having to copy&paste Mainnet and Goerli in several locations? Could we address this in a different way such as renaming `DEFAULT_NETWORKS`?